lib/graphlient/client.rb

```diff
@@ -51,11 +54,17 @@ def http(&block)
     end
 
     def schema
-      @schema ||= Graphlient::Schema.new(http, schema_path)
+      @schema ||= options[:schema] || Graphlient::Schema.new(http, schema_path)
     end
 
     private
 
+    def raise_error_if_invalid_configuration
```

Comment:
nit: since this method raises, rename it to `raise_error_if_invalid_configuration!`

Reply:
Makes sense.